pkg/agent/route/route_linux.go

```diff
@@ -152,7 +152,79 @@ func (c *Client) syncIPInfra() {
 		klog.Errorf("Failed to sync iptables: %v", err)
 		return
 	}
-	klog.V(3).Infof("Successfully synced node iptables")
+	if err := c.syncRoutes(); err != nil {
+		klog.Errorf("Failed to sync routes: %v", err)
+	}
+	klog.V(3).Infof("Successfully synced node iptables and routes")
+}
+
+func (c *Client) syncRoutes() error {
+	routeList, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return err
+	}
+	routeMap := make(map[string]*netlink.Route)
+	for i := range routeList {
+		r := &routeList[i]
+		if r == nil || r.Dst == nil {
+			continue
+		}
+		routeMap[r.Dst.String()] = r
+	}
+	routes := []*netlink.Route{}
+	c.nodeRoutes.Range(func(_, v interface{}) bool {
+		for _, route := range v.([]*netlink.Route) {
+			routes = append(routes, route)
+		}
+		return true
+	})
+	for _, route := range routes {
```

Comment:
Could this handling just be in the loop of `c.nodeRoutes.Range`? It doesn't seem necessary to construct a slice first